src/ansys/mechanical/core/embedding/app.py

```diff
@@ -194,9 +196,35 @@ def save(self, path=None):
         else:
             self.DataModel.Project.Save()
 
-    def save_as(self, path):
-        """Save the project as."""
-        self.DataModel.Project.SaveAs(path)
+    def save_as(self, path: str, overwrite: bool = False):
+        """Save the project as a new file.
+
+        If the `overwrite` flag is enabled, the current saved file is temporarily moved
+        to a backup location. The new file is then saved in its place. If the process fails,
+        the backup file is restored to its original location.
+
+        Parameters
+        ----------
+        path: int, optional
+            The path where file needs to be saved.
+        overwrite: bool, optional
+            Whether the file should be overwritten if it already exists (default is False).
+        """
+        if overwrite:
+            if os.path.exists(path):
```

Comment:
I think you're missing an else branch here. If the path doesn't exist, it should do the same thing as if overwrite is not set to true.